Automattic/wp-calypso
# Plugins Management: Use the new endpoint for fetching all site plugins

#### Proposed Changes

In https://github.com/Automattic/wp-calypso/pull/61803, @enejb did a fantastic job using `react-query` for some plugin endpoints, but IMO we should try to split such refactor into much smaller pieces to avoid the possibility of introducing a regression.

Migrating all plugins to react-query would require us to:
- Research if we can replace Redux completely. In a quick chat, @tyxla pointed out that there might be some dynamics in the UI (e.g., clicking "Activate plugin" would be something we currently can't track on the server side, but it's available in the store).
- Define the phases of how this should happen.
- Work on small incremental steps towards the goal. 

What about this PR?

As properly migrating to react-query would require more time, this changeset introduces a new query component, `QueryAllJetpackPlugins` that is used in all places where we need to fetch installed plugins _for all Jetpack sites._ 
The existing `<QueryJetpackPlugins />` component will still be used for fetching plugins for _a single Jetpack site_.

We found some inconsistencies, which were documented in: pbtFFM-29g-p2

**Note:** I'm working on some tests, but I'd like to get this code reviewed while working on them.

#### Testing Instructions
- Checkout PR locally
- Run `yarn start`
- When the build is ready, run `yarn start-jetpack-cloud-p`
- Visit http://calypso.localhost:3000/plugins/manage and verify that the screen loads as expected (hopefully more instantly as well)
- Verify that results match the ones from the production environment. If they don't match, this shouldn't be directly related to this PR, but would be good to investigate the case further.
- Verify that bulk actions (install, activate, enable auto-update, update and remove plugins) still work as expected.
- Visit http://calypso.localhost:3000/plugins/akismet and verify that the screen loads as expected.
- Visit http://calypso.localhost:3000/plugins/browse/popular and verify that the screen loads as expected.
- Visit http://jetpack.cloud.localhost:3001/plugins/manage and verify that the screen loads as expected (and matches what you see on Calypso Blue)
- Visit http://jetpack.cloud.localhost:3001/plugins/akismet and verify that the screen loads as expected.
- Visit http://jetpack.cloud.localhost:3001/plugins/manage/<SITE> and verify that the screen loads as expected.

#### Pre-merge Checklist






Related to 1202518759611394-as-1202534044814282
